internal/controller/certificaterequests/OWNERS

```diff
@@ -0,0 +1,4 @@
+filters:
+  "^apply(_test)?\\.go$":
```

Comment:
So this ensures that you are a required reviewer for both `apply.go` and `apply_test.go`. What regex syntax is this, why does it need `\\` before `.` (as opposed to `\`)?

Reply:
I don't know, I have followed the example from [here](https://github.com/kubernetes/kubernetes/blob/a266805f33197ba17577e2e71511f68ce9db57eb/OWNERS#L25). Perhaps this is wrong for this case, I'm not sure and don't know how to test it.

Reply:
The owners file configs are being parsed by Prow [here](https://github.com/kubernetes/test-infra/blob/master/prow/repoowners/repoowners.go#L534-L544) (the `pattern` in this case would be the regex).

Required owners are being determined [here](https://github.com/kubernetes/test-infra/blob/master/prow/repoowners/repoowners.go#L846-L852) where `path` is the file name and further the regex is matched against relative path [here](https://github.com/kubernetes/test-infra/blob/master/prow/repoowners/repoowners.go#L796).
So, `^apply(_test)?\\.go$` won't match against something like `internal/controller/certificaterequests` and we should remove the `^`, but otherwise it will match. (I realized that `\\` is needed because it's a [go thing](https://stackoverflow.com/questions/6770898/unknown-escape-sequence-error-in-go)).

Tried it in Go playground as I like regular expressions :) https://go.dev/play/p/3D6XM98Lty2